### DISCUSSION TITLE
h5fortran error

### DISCUSSION THREAD

**Discussion by EmilioEchevarria** [modern-fortran/neural-fortran#159] [Q&A]
Hello all, 

I recently came across neural-fortran, and it seems to be exactly what I need. I'm interested in using neural networks built in python with Keras, load them and use them in a Fortran-based numerical model (WAVEWATCH III), so a big thank you for making this available to everyone!

I followed the steps for the installation with CMake, and it all seems to work well. I'm able to get the tests in the build/bin directory. I can execute `simple`, `sine`, `dense_mnist`, `cnn_mnist`, but not `dense_from_keras` or `cnn_from_keras` which are the ones I'm more interested in. I'm getting this error: 

```
ERROR STOP h5fortran:open: HDF5 library initialize
```

I didn't install h5fortran myself, I assumed this would be done automatically, and I see a `h5fortran.mod` in build/include. 
 
I'm really a beginner with Fortran, so this is probably a silly issue, but if anyone can point me in the right direction here to try and solve this, that would be much appreciated! :) 

Cheers,
Emilio

**Comment by milancurcic** [comment]
Hi Emilio,

I don't know why this happens. It seems to occur in the following section of h5fortran:

https://github.com/geospace-code/h5fortran/blob/671dfa25d5c4820e88b1dba6147b471ac0f8459d/src/utils.f90#L73-L80

Can you open an issue about this on https://github.com/geospace-code/h5fortran and specify your environment details, specifically: OS, C and Fortran compilers and versions, HDF5 version.

Thank you!

**Reply by EmilioEchevarria** [reply]
Hi Milan, 

Thank you so much, I just saw this https://github.com/geospace-code/h5fortran/issues/42#issuecomment-1648669025, I guess my hdf5 version was the problem (1.14.0). I installed hdf5-v1.12.0 instead and now it works :)
